Text/CustomEntitySearch/CustomEntitySearch.cs

```diff
@@ -95,5 +149,181 @@ public static class CustomEntitySearch
+                                new Entity
+                                {
+                                    Category = "customEntity",
+                                    Value = text.Substring(startPointersInText[i], endPointersInText[endofMatchInTextPointer[i]] - startPointersInText[i] + 1),
+                                    Offset = startPointersInText[i],
+                                    Confidence = minLevenshteinDistance[i]
+                                });
+                        entitiesFound.Add(checkMatch);
+                    }
+                }
+
+
+            }
+        }
+
+        private static double DamerauLevenshteinCalculation(string text, string checkMatch)
+        {
+            double[,] dynamicDistanceCalc = new double[text.Length + 1, checkMatch.Length + 1];
+            double substitutionCost = -1;
+            double accentAddition = 1;
+            for (int currTextIndex = 0; currTextIndex <= text.Length; currTextIndex++)
+                dynamicDistanceCalc[currTextIndex, 0] = currTextIndex;
+            for (int currWordIndex = 0; currWordIndex <= checkMatch.Length; currWordIndex++)
+                dynamicDistanceCalc[0, currWordIndex] = currWordIndex;
+
+            for (int currTextIndex = 0; currTextIndex < text.Length; currTextIndex++)
+            {
+                for (int currWordIndex = 0; currWordIndex < checkMatch.Length; currWordIndex++)
+                {
+                    if (text[currTextIndex].Equals(checkMatch[currWordIndex]))
+                        substitutionCost = 0;
+                    else if (String.Compare(checkMatch[currWordIndex].ToString(), text[currTextIndex].ToString(),
+                                    CultureInfo.CurrentCulture, CompareOptions.IgnoreNonSpace) == 0)
+                        substitutionCost = .5;
+                    else
+                        substitutionCost = 1;
+                    if (checkMatch[currWordIndex].IsAccent() ^ text[currTextIndex].IsAccent())
+                        accentAddition = 0.5;
+                    dynamicDistanceCalc[currTextIndex + 1, currWordIndex + 1] = Math.Min(
+                        Math.Min(dynamicDistanceCalc[currTextIndex, currWordIndex + 1] + accentAddition, // deletion
+                        dynamicDistanceCalc[currTextIndex + 1, currWordIndex] + accentAddition), // insertion
+                        dynamicDistanceCalc[currTextIndex, currWordIndex] + substitutionCost); // substitution
+                    if (currTextIndex > 0 && currWordIndex > 0 && text[currTextIndex].Equals(checkMatch[currWordIndex - 1]) && checkMatch[currWordIndex].Equals(text[currTextIndex - 1]))
+                        dynamicDistanceCalc[currTextIndex + 1, currWordIndex + 1] = Math.Min(dynamicDistanceCalc[currTextIndex + 1, currWordIndex + 1], 
+                            dynamicDistanceCalc[currTextIndex - 1, currWordIndex - 1] + substitutionCost); // transposition
+                }
```

Comment:
this looks like you're having like 

aaazxbbb
aaaxzbbb

count as a diff of 1? Am i reading that right?

Reply:
synced offline, yes this is by design

---
In reply to: [322980495](https://github.com/Azure-Samples/azure-search-power-skills/pull/5#discussion_r322980495) [](ancestors = 322980495)